android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt

```diff
@@ -155,52 +185,107 @@ private fun SurveyQuestionsContent(questions: List<QuestionContentUiModel>) {
+        val isLastPage = pagerState.currentPage == pagerState.pageCount - 1
         Button(
-            onClick = {},
+            onClick = {
+                coroutineScope.launch {
+                    pagerState.animateScrollToPage(pagerState.currentPage + 1)
+                }
+            },
             colors = ButtonDefaults.buttonColors(
-                backgroundColor = Color.Transparent
+                backgroundColor = if (isLastPage) Color.White else Color.Transparent
             ),
             elevation = null,
             contentPadding = PaddingValues(0.dp),
+            shape = if (isLastPage) RoundedCornerShape(10.dp) else MaterialTheme.shapes.small,
             modifier = Modifier.constrainAs(nextButton) {
                 bottom.linkTo(parent.bottom, margin = 54.dp)
                 end.linkTo(parent.end, margin = 20.dp)
             }
         ) {
-            Image(
-                painter = painterResource(id = R.drawable.ic_survey_detail_arrow),
-                contentDescription = null,
-                contentScale = ContentScale.FillWidth,
-                modifier = Modifier.size(56.dp),
-            )
+            if (isLastPage) {
+                Text(
+                    text = stringResource(id = R.string.submit_button),
+                    color = Color.Black,
+                    style = Typography.subtitle2,
+                    textAlign = TextAlign.Center,
+                    modifier = Modifier
+                        .width(120.dp)
+                        .height(56.dp)
+                        .wrapContentHeight()
+                )
+            } else {
+                Image(
+                    painter = painterResource(id = R.drawable.ic_survey_detail_arrow),
+                    contentDescription = null,
+                    contentScale = ContentScale.FillWidth,
+                    modifier = Modifier.size(56.dp)
+                )
+            }
         }
     }
 }
 
+@Composable
+private fun Answer(type: QuestionDisplayType, onAnswerClick: (Int) -> Unit) {
+    when (type) {
+        QuestionDisplayType.STAR -> EmojiRatingAnswer(
+            emojis = MutableList(5) {
+                { Emoji(name = STAR_EMOJI) }
+            },
+            onIndexChange = onAnswerClick
+        )
+        QuestionDisplayType.HEART -> EmojiRatingAnswer(
+            emojis = MutableList(5) {
+                { Emoji(name = HEART_EMOJI) }
+            },
+            onIndexChange = onAnswerClick
+        )
+        QuestionDisplayType.SMILEY -> EmojiRatingAnswer(
+            emojis = listOf(
+                { Emoji(name = POUTING_FACE_EMOJI) },
+                { Emoji(name = CONFUSED_FACE_EMOJI) },
+                { Emoji(name = NEUTRAL_FACE_EMOJI) },
+                { Emoji(name = SLIGHTLY_SMILING_FACE_EMOJI) },
+                { Emoji(name = GRINNING_FACE_WITH_SMILING_EYES_EMOJI) }
+            ),
+            onIndexChange = onAnswerClick
+        )
```

Comment:
@suho Please kindly check the style for highlighted emoji style 😄

Reply:
Nice catch! Fixed in https://github.com/suho/kmm-ic/pull/116/commits/949ea88c53e236ecb61123ff53995043f6d52189